optimade/models/structures.py

```diff
@@ -64,12 +69,12 @@ class Species(BaseModel):
 
     """
 
-    name: str = Field(
+    name: str = StrictField(
```

Comment:
Should these fields also be `OptimadeField`s? This is perhaps also relevant for the fields in other "sub"-models.

Reply:
Yeah, good spot, it's pretty clear in the spec here that these models have defined support levels for the subfields

Reply:
Although, the dict itself is only a SHOULD, so I guess that maxes out all of the support levels of the subfields to SHOULD too (they are listed as "REQUIRED") in the spec...

Reply:
Well I see it more as _if_ `species` is included it `MUST` include these fields... so it depends on what happens if these fields are `MUST` in the validation, while `species` is "only" `SHOULD`?

Reply:
Also, this may be the case for non-entry listing fields, i.e. that we need to use `OptimadeField`. But I consider that update fine for a separate PR if necessary.

Reply:
> Well I see it more as _if_ `species` is included it `MUST` include these fields... so it depends on what happens if these fields are `MUST` in the validation, while `species` is "only" `SHOULD`?

Makes sense, as long as that's how we interpret it!

> Also, this may be the case for non-entry listing fields, i.e. that we need to use `OptimadeField`. But I consider that update fine for a separate PR if necessary.

Yeah, wasn't sure about this either... queryable doesn't really make sense for non-entry fields right? and SupportLevel is handled more directly by OpenAPI's `required` (I don't think any of the non-entry listing fields are SHOULDs, are they? :worried: )

Reply:
Yeah, let's just hold out on the non-entry listing fields for now.
And the level of SHOULD and MUSTS - yeah, I think we should try and make it match the specification in this meaning and transfer that understanding to the validator as well, i.e., the support level conditions are only valid within it's own "tier".